src/OSPSuite.Core/Domain/ContainerExtensions.cs

```diff
@@ -185,12 +186,26 @@ public static void AddChildren(this IContainer container, IEnumerable<IEntity> c
          return container?.GetSingleChild<TChild>(x => true);
       }
 
+      /// <summary>
+      ///    Returns the formula,  using formula referencing and corresponding path entries fulfilling the given
+      ///    <paramref name="predicate" /> within the container
+      /// </summary>
+      /// <param name="container">Container where paths used in formula will be checked against predicate</param>
+      /// <param name="predicate">The predicate to check formula path against</param>
+      /// <returns></returns>
+      public static IEnumerable<EntityFormulaPath> GetPathsReferencing(this IContainer container, Func<FormulaUsablePath, bool> predicate)
+      {
+         return getAllPathReferencing<IUsingFormula>(container, predicate, x => x.Formula)
+            .Union(getAllPathReferencing<IParameter>(container, predicate, x => x.RHSFormula));
+      }
 
-      public static IEnumerable<(IUsingFormula usingFormula, FormulaUsablePath path)> GetPathsReferencing(this IContainer container, Func<FormulaUsablePath, bool> predicate)
+      private static IEnumerable<EntityFormulaPath> getAllPathReferencing<T>(IContainer container, Func<FormulaUsablePath, bool> predicate, Func<T, IFormula> formulaFunc) where T : class, IEntity
       {
-         return container.GetAllChildren<IUsingFormula>(x => x.Formula.ObjectPaths.Any(predicate))
-            .SelectMany(usingFormula => usingFormula.Formula.ObjectPaths.Where(predicate)
-               .Select(path => (usingFormula, path)));
+         return container.GetAllChildren<T>(x => formulaFunc(x) != null)
```

Comment:
@Yuri05 @rwmcintosh It's the same code, Only I specify now where to get the formula from